### ISSUE TITLE
Keine Icons unter Seitenstruktur

### ISSUE THREAD

**Issue by NicolasGoerlach** [menatwork-ia/clipboard#15]
Hi, 
sobald ich das Clipboard bei einem User aktiviere passiert mit den Icons der Seitenstruktur folgendes:

![clipboard-bug](https://cloud.githubusercontent.com/assets/8327905/19291286/901b7462-9014-11e6-99d9-41adf52c40b5.png)

Wenn ich das Clipboard deaktiviere, sind alle Icons wieder da.

Version: menatwork/clipboard 2.0.6
Über Composer installiert

**Comment by johannes-bittner** [comment]
Kann ich bestätigen für den gleichzeitigen Einsatz mit changelanguage (Contao 3.5.18, jeweils aktuelle Versionen über Composer): Keine Icons in Seiten und Artikelstruktur. Zudem sind übergeordnete Seiten nicht mehr klickbar, um die Anzeigetiefe einzuschränken:

![artikel-cl-aktiv](https://cloud.githubusercontent.com/assets/1597697/19930858/8fb3e084-a10a-11e6-92c4-c63db55a4c9f.jpg)
(Clipboard *und* Changelanguage aktiv: Icons weiß, nicht klickbar)

![artikel-cl-inaktiv](https://cloud.githubusercontent.com/assets/1597697/19930878/ab851738-a10a-11e6-9e07-8422c06bfe45.jpg)
(Clipboard *oder* Changelanguage aktiv: alles normal)

**Comment by zonky2** [comment]
@NicolasGoerlach und  @johannes-bittner bitte bei ChangeLanguage die develop-Version installieren bzw. testen - hier gab es eine Anpassung für die bessere Zusammenarbeit mit anderen Erweiterungen
https://github.com/terminal42/contao-changelanguage/blob/develop/library/Terminal42/ChangeLanguage/EventListener/DataContainer/MissingLanguageIconListener.php#L36-L53

**Comment by NicolasGoerlach** [comment]
@zonky2 
terminal42/contao-changelanguage dev-develop (a3c1d2ac) 
installiert, leider immer noch der selbe Fehler.

**Comment by zonky2** [comment]
schade :-(
dann muss MAW doch mal ran...

**Comment by zonky2** [comment]
o.k. - ich habe auch eine Erweiterung, die an die Labels im Seitenbaum etwas "anklebt" - das gleiche Problem...

m.E. muss an der Stelle 
https://github.com/menatwork-ia/clipboard/blob/master/system/modules/clipboard/drivers/DC_Clipboard.php
das Clipboard überarbeitet werden, dass auch andere Label-Callbacks zum Zuge kommen können so wie hier 
https://github.com/terminal42/contao-changelanguage/commit/83281870e4c08476fcde02e1bdd67d9cc40b9700

**Comment by Toflar** [comment]
Daran liegt es nicht. Es liegt an dem: https://github.com/menatwork-ia/clipboard/issues/19
Der DC ist einfach komplett outdated. Viele nutzen Closures für die Callbacks, welche in Contao 3.2 eingeführt wurden (über 3 Jahre her). Das kann der DC_Clipboard nicht.

**Comment by zonky2** [comment]
ja, habe mir das "Konstrukt" nochmal genauer angesehen... da müsste "man" wahrscheinlich ordentlich die Ärmel hoch krempeln

**Comment by zonky2** [comment]
... schade - war mal ein tolles Tool - auch um Content von einer Seite zu einer anderen zu schaufeln

mit Contao 4 wird das (auch) den "Heldentod" sterben :-(

**Comment by BugBuster1701** [comment]
Der Pullrequest #22  würde ja schon mal die Closures unterstützen wenn ich das richtig sehe.

**Comment by zonky2** [comment]
habe letztes Jahr bei MAW im Projekt MCW einige Tickets aus 2013 aufgearbeitet... d.h. es ist i.E. mit 2020 zu rechnen ;-)
